src/query/functions/utils/heap.go

```diff
@@ -137,14 +152,27 @@ func (fh *FloatHeap) Reset() {
 }
 
 // Flush flushes the float heap and resets it. Does not guarantee order.
-func (fh FloatHeap) Flush() []ValueIndexPair {
+func (fh *FloatHeap) Flush() []ValueIndexPair {
 	values := fh.floatHeap.heap
 	fh.Reset()
 	return values
 }
 
+// FlushInOrder flushes the float heap and returns values in order.
+func (fh *FloatHeap) FlushInOrder() []ValueIndexPair {
+	result := make([]ValueIndexPair, fh.Len())
+
+	for i := len(result) - 1; i >= 0; i-- {
+		if e, ok := fh.Pop(); ok {
+			result[i] = e
+		}
+	}
+
+	return result
+}
```

Comment:
Rather than having our own heapsort here, probably easier to use internal sort; e.g. this passes all tests:

```golang
// OrderedFlush flushes the float heap and returns values in order.
func (fh FloatHeap) OrderedFlush() []ValueIndexPair {
	flushed := fh.Flush()
	if fh.isMaxHeap {
		sort.Slice(flushed, func(i, j int) bool {
			return fh.floatHeap.less(flushed[i], flushed[j])
		})
	}

	return flushed
}
```

Better yet may be to take a `Flush(ordered bool)` and key ordering based on that